packages/react/dialog/src/Dialog.tsx

```diff
@@ -396,19 +387,20 @@ const DialogContentImpl = React.forwardRef<DialogContentImplElement, DialogConte
           <DismissableLayer
             role="dialog"
             id={context.contentId}
-            aria-describedby={ariaDescribedBy || context.descriptionId}
-            // If `aria-label` is set, ensure `aria-labelledby` is undefined as to avoid confusion.
-            // Otherwise fallback to an explicit `aria-labelledby` or the ID used in the
-            // `DialogTitle`
-            aria-labelledby={ariaLabel ? undefined : ariaLabelledBy || context.titleId}
-            aria-label={ariaLabel || undefined}
+            aria-describedby={context.descriptionId}
+            aria-labelledby={context.titleId}
             data-state={getState(context.open)}
             {...contentProps}
             ref={composedRefs}
             onDismiss={() => context.onOpenChange(false)}
           />
         </FocusScope>
-        {process.env.NODE_ENV === 'development' && <LabelWarning contentRef={contentRef} />}
+        {['development', 'test'].includes(process.env.NODE_ENV!) && (
```

Comment:
I've always wondered about this. We do `process.env.NODE_ENV === 'development'` everywhere but should we be doing `process.env.NODE_ENV !== 'production'` instead?